erpnext/accounts/doctype/subscription/subscription.py

```diff
@@ -523,7 +525,7 @@ def process_for_active(self):
 		if getdate() > getdate(self.current_invoice_end) and self.is_prepaid_to_invoice():
 			self.update_subscription_period(add_days(self.current_invoice_end, 1))
 
-		if not self.is_current_invoice_generated() and (self.is_postpaid_to_invoice() or self.is_prepaid_to_invoice()):
+		if not self.is_current_invoice_generated(self.current_invoice_start, self.current_invoice_end) and (self.is_postpaid_to_invoice() or self.is_prepaid_to_invoice()):
```

Comment:
Same change needed in `process_for_past_due_date`

Reply:
@nextchamp-saqib Fixed. Check now